### PR TITLE
SLI-930 Add user interactions to review the security hotspot

### PR DESCRIPTION
**Pull request by eray-felek-sonarsource**
Please review our [contribution guidelines](https://github.com/SonarSource/sonarlint-intellij/blob/master/contributing.md).

And please ensure your pull request adheres to the following guidelines: 

- [ ] Please explain your motives to contribute this change: what problem you are trying to fix, what improvement you are trying to make
- [ ] Use the following formatting style: [SonarSource/sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style)
- [ ] Provide a unit test for any code you changed
- [ ] If there is a [JIRA](http://jira.sonarsource.com/browse/SLI) ticket available, please make your commits and pull request start with the ticket ID (SLI-XXXX)

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/ui/ruledescription/RuleHeaderPanel.kt [hunk 1]
```diff
@@ -64,6 +68,12 @@ class RuleHeaderPanel : JBPanel<RuleHeaderPanel>(FlowLayout(FlowLayout.LEFT)) {
         add(ruleKeyLabel.apply {
             border = JBUI.Borders.emptyLeft(10)
         }, HorizontalLayout.CENTER)
+        changeStatusButton.action = object : AbstractAction() {
```
Comment (nquinquenel) [inline src/main/java/org/sonarlint/intellij/ui/ruledescription/RuleHeaderPanel.kt:71]: I think you could simply use `addActionListener()` in this case

File: src/main/java/org/sonarlint/intellij/actions/OpenSecurityHotspotInDialogAction.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.project.Project
+
+class OpenSecurityHotspotInDialogAction : AbstractSonarAction(
+    "Open In Dialog",
+    "Open security hotspot in dialog",
+    null
+) {
+    override fun updatePresentation(e: AnActionEvent, project: Project) {
```
Comment (nquinquenel) [inline src/main/java/org/sonarlint/intellij/actions/OpenSecurityHotspotInDialogAction.kt:11]: I don't think you have to override `updatePresentation`, this is mostly used when text is dynamic (having to adjust title on SQ or SC for example)

File: src/main/java/org/sonarlint/intellij/ui/ruledescription/RuleHeaderPanel.kt [hunk 3]
```diff
@@ -93,6 +104,9 @@ class RuleHeaderPanel : JBPanel<RuleHeaderPanel>(FlowLayout(FlowLayout.LEFT)) {
             setCopyable(true)
             background = SonarLintIcons.colorsByProbability[vulnerabilityProbability]
         }
+
+        changeStatusButton.isVisible = true
+        changeStatusButton.text = "Change status"
```
Comment (damien-urruty-sonarsource) [inline src/main/java/org/sonarlint/intellij/ui/ruledescription/RuleHeaderPanel.kt:109]: I think you could set this text only once in the `init`

File: src/main/java/org/sonarlint/intellij/actions/OpenSecurityHotspotInDialogAction.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package org.sonarlint.intellij.actions
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.project.Project
+
+class OpenSecurityHotspotInDialogAction : AbstractSonarAction(
```
Comment (damien-urruty-sonarsource) [inline src/main/java/org/sonarlint/intellij/actions/OpenSecurityHotspotInDialogAction.kt:6]: I would suggest to use a more precise name, what is really the purpose of the action ? Something like `OpenHotspotStatusChangeDialogAction`

**Review by damien-urruty-sonarsource — APPROVED** [review]
LGTM :+1:

**Review by nquinquenel — APPROVED** [review]
LGTM!

For the build issue you can use `gradle licenseFormat` to generate license header on each new file :+1:

**Pull request merged**